Automattic/woocommerce-payments
# Update _charge_id in metadata

Fixes #3857 

#### Changes proposed in this Pull Request

Some sites face issues with UPE payments as described at #3857
1) Refund button does not appear
2) In the Transactions screen, order id column shows N/A
3) In the Order screen, the link is not present on the Payment Intent id.
Payment is completed and is correct on Order Notes, Stripe dashboard and WCPay. 

On testing, this seems to happen when the function `maybe_process_upe_redirect` in `UPE_Payment_Gateway` does not run. When this happens, but the `process_webhook_payment_intent_succeeded` runs, all values are updated except `_charge_id` metadata on the order. To fix this, `_charge_id` is updated in `process_webhook_payment_intent_succeeded`. 

#### Testing instructions
1. Setup a JN site or a local site with webhook processing.
2. Comment the line 107 in `includes/payment-methods/class-upe-payment-gateway.php`
`add_action( 'wp', [ $this, 'maybe_process_upe_redirect' ] );`
3. On develop branch, observe the symptoms of the problem as described above. Observe that all metadata of the order is present except _charge_id.
4. On `fix/3857-update-charge-id` branch, observe that `_charge_id` is updated. Order id is present in Transactions, refund button works, etc.

-------------------


**Post merge**
